include/vcpkg/base/downloads.h

```diff
@@ -27,46 +27,67 @@ namespace vcpkg::Downloads
     // Returns url that was successfully downloaded from
     std::string download_file(Files::Filesystem& fs,
                               View<std::string> urls,
+                              View<std::string> headers,
                               const fs::path& download_path,
                               const std::string& sha512);
 
     void download_file(Files::Filesystem& fs,
                        const std::string& url,
+                       View<std::string> headers,
                        const fs::path& download_path,
                        const std::string& sha512);
 
+    View<std::string> azure_blob_headers();
+
     std::vector<int> download_files(Files::Filesystem& fs, View<std::pair<std::string, fs::path>> url_pairs);
-    int put_file(const Files::Filesystem&, StringView url, const fs::path& file);
-    std::vector<int> url_heads(View<std::string> urls);
+    ExpectedS<int> put_file(const Files::Filesystem&, StringView url, View<std::string> headers, const fs::path& file);
+    std::vector<int> url_heads(View<std::string> urls, View<std::string> headers);
+
+    struct DownloadManagerConfig
```

Comment:
I would prefer whatever you're testing just return one of these instead of needing an escape hatch to get this back out.